airflow/www/views.py

```diff
@@ -752,18 +752,22 @@ def xcom(self, session=None):
     @has_access
     @action_logging
     def run(self):
+        """Trigger a run of a task, starting a task instance."""
         dag_id = request.args.get('dag_id')
         task_id = request.args.get('task_id')
         origin = request.args.get('origin')
-        dag = dagbag.get_dag(dag_id)
-        task = dag.get_task(task_id)
-
         execution_date = request.args.get('execution_date')
-        execution_date = pendulum.parse(execution_date)
         ignore_all_deps = request.args.get('ignore_all_deps') == "true"
         ignore_task_deps = request.args.get('ignore_task_deps') == "true"
         ignore_ti_state = request.args.get('ignore_ti_state') == "true"
 
+        dag_model = DagModel.get_dagmodel(dag_id)
+        if not dag_model:
+            flash("DAG '{0}' not found.".format(dag_id), "error")
+            return redirect("/")
```

Comment:
Eek. This is wrong (including in all the other ones we've already merged.). We shouldn't be hard-coding URLs as it will be wrong if base_url is specified. `url_for('Airflow.index')`

Reply:
Given this is quite a common pattern it might be worth pulling out in to a `dag_from_model_or_redirect()` helper function somewhere?

Reply:
My bad, didn't know about the base_url thing. I've extracted a method `_get_dag_model_or_redirect()` to do this and fixed the redirect.